### PR TITLE
Homework #20

### PR DESCRIPTION
**Pull request by vetalbunyak**

### CODE REVIEW

File: src/main/java/ithillel/ua/glovo/controller/OrderController.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+import ithillel.ua.glovo.model.Order;
+import ithillel.ua.glovo.model.Product;
+import ithillel.ua.glovo.service.OrderService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@RequestMapping("/orders")
```
Comment (mkharkhuta) [inline src/main/java/ithillel/ua/glovo/controller/OrderController.java:11]: "/" can be omitted (optional)

File: src/main/java/ithillel/ua/glovo/controller/OrderController.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+public class OrderController {
+    @Autowired
+    private OrderService orderService;
+
+    @GetMapping("/{id}")
```
Comment (mkharkhuta) [inline src/main/java/ithillel/ua/glovo/controller/OrderController.java:16]: "/" can be omitted (optional) and the same for other

File: src/main/java/ithillel/ua/glovo/controller/OrderController.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    public ResponseEntity<Order> getOrder(@PathVariable Long id) {
+        Order order = orderService.getOrderById(id);
+        return ResponseEntity.ok(order);
+    }
+
+    @PostMapping
+    public ResponseEntity<Order> addOrder(@RequestBody Order order){
+        Order newOrder = orderService.addOrder(order);
+        return ResponseEntity.ok(newOrder);
```
Comment (mkharkhuta) [inline src/main/java/ithillel/ua/glovo/controller/OrderController.java:25]: what if the order with the same ID already exit?

File: src/main/java/ithillel/ua/glovo/model/Order.java [hunk 4]
```diff
@@ -0,0 +1,13 @@
+package ithillel.ua.glovo.model;
+
+import lombok.Getter;
+import lombok.Setter;
+
+import java.util.List;
+@Getter
+@Setter
+public class Order {
+    private  Long id;
```
Comment (mkharkhuta) [inline src/main/java/ithillel/ua/glovo/model/Order.java:10]: too many spaces, please refactor

File: src/main/java/ithillel/ua/glovo/service/OrderServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+        return order;
+    }
+
+    @Override
+    public Order addProductToOrder(Long orderId, Product product) {
+        Order order = orders.get(orderId);
+        if (order != null) {
+            order.getProducts().add(product);
+        }
+        return order;
```
Comment (mkharkhuta) [inline src/main/java/ithillel/ua/glovo/service/OrderServiceImpl.java:39]: it's better to send a message back, like: order with id {} not found

**Review by mkharkhuta — COMMENTED** [review]
good job 👍

**Review by shaitanaleksander — APPROVED** [review]

File: src/main/java/ithillel/ua/glovo/controller/OrderController.java [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+        if (order != null) {
+            return new ResponseEntity<>(order, HttpStatus.OK);
+        }
+        return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+    }
+
+    @PostMapping
+    public ResponseEntity<Order> addOrder(@RequestBody Order order) {
+        Order newOrder = orderService.addOrder(order);
+        return new ResponseEntity<>(newOrder, HttpStatus.CREATED);
```
Comment (mkharkhuta) [inline src/main/java/ithillel/ua/glovo/controller/OrderController.java:30]: The bad request or another status should be returned in case of failure

**Pull request merged**